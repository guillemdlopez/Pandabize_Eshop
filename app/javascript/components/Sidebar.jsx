import React from "react";
import { useHistory } from "react-router";
import PropTypes from "prop-types";

const Sidebar = ({ avatar, first_name, last_name, orders }) => {
  const history = useHistory();

  const goToHomePage = () => {
    history.push("/");
  };

  return (
    <div className="sidebar d-flex align-items-center">
      <img src={avatar} className="avatar-md" />
      <div className="user-info">
        <ul className="sidebar-list ml-3">
          <li className="full_name-item">
            <strong>Full Name: </strong>
            {`${first_name} ${last_name}`}
          </li>
          <li className="orders-item">
            <strong>Orders:</strong> {orders?.length}
          </li>
          <li>
            <button
              className="btn btn-outline-success btn-sm"
              onClick={goToHomePage}
            >
              Continue shopping
            </button>
          </li>
        </ul>
      </div>
    </div>
  );
};

Sidebar.propTypes = {
  avatar: PropTypes.string,
  first_name: PropTypes.string,
  last_name: PropTypes.string,
  orders: PropTypes.array,
};

export default Sidebar;
