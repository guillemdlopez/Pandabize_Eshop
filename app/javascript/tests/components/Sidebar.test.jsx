import { shallow } from "enzyme";
import React from "react";
import { useHistory } from "react-router-dom";
import Sidebar from "../../components/Sidebar";

describe("Testing the <Sidebar/> component", () => {
  const sidebarProps = {
    avatar: "avatar.jpg",
    first_name: "Guillem",
    last_name: "Delás",
    orders: [
      {
        id: 19,
        bicycle_id: 26,
        amount: 13.1,
        user_id: 10,
        created_at: "2021-03-08T23:54:06.646Z",
        updated_at: "2021-03-08T23:54:06.646Z",
        bicycle: "New School XC",
        photo:
          "http://res.cloudinary.com/doker55/image/upload/jbwjj6i6r1f2oz551vxojkmtb42c.jpg",
        customizations: [
          {
            id: 148,
            name: "Wheel size",
            value: "16 inches",
            price: 6.15,
            bicycle_id: 26,
            created_at: "2021-03-08T19:34:57.247Z",
            updated_at: "2021-03-08T19:34:57.247Z",
          },
          {
            id: 168,
            name: "Rim color",
            value: "Blue",
            price: 6.95,
            bicycle_id: 26,
            created_at: "2021-03-08T19:34:57.404Z",
            updated_at: "2021-03-08T19:34:57.404Z",
          },
        ],
      },
    ],
  };

  const wrapper = shallow(<Sidebar {...sidebarProps} history={history} />);
  test("should render the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should display the full name of the current user", () => {
    expect(
      wrapper.find(".full_name-item").text().includes("Guillem Delás")
    ).toBe(true);
  });

  test("should display the number of orders of the current_user", () => {
    expect(wrapper.find(".orders-item").text()).toBe("Orders: 1");
  });
});
