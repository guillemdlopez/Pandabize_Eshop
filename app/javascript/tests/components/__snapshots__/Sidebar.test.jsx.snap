// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <Sidebar/> component should render the component correctly 1`] = `
<div
  className="sidebar d-flex align-items-center"
>
  <img
    className="avatar-md"
    src="avatar.jpg"
  />
  <div
    className="user-info"
  >
    <ul
      className="sidebar-list ml-3"
    >
      <li
        className="full_name-item"
      >
        <strong>
          Full Name: 
        </strong>
        Guillem Del√°s
      </li>
      <li
        className="orders-item"
      >
        <strong>
          Orders:
        </strong>
         
        1
      </li>
      <li>
        <button
          className="btn btn-outline-success btn-sm"
          onClick={[Function]}
        >
          Continue shopping
        </button>
      </li>
    </ul>
  </div>
</div>
`;
